File: Ders2/app/src/main/java/com/xzcorp/ders2/MainActivity.java
package com.xzcorp.ders2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    TextView mesajDegisken;
    Button buttonDegisken;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        buttonDegisken = (Button)findViewById(R.id.kaydet);
        mesajDegisken = (TextView)findViewById(R.id.mesaj);

        buttonDegisken.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mesajDegisken.setText("Text Değişti");
            }
        });
        buttonDegisken.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                Toast.makeText(MainActivity.this,"Uzun Bastın",Toast.LENGTH_LONG).show();
                return false;
            }
        });
        buttonDegisken.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                mesajDegisken.setText("Hover");
            }
        });
    }
}

File: Ders16/app/src/main/java/com/xzcorp/ders16/MainActivity.kt
package com.xzcorp.ders16

import android.graphics.Color
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.LinearLayout
import android.widget.ListView
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.RatingBar
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    lateinit var secim:Button
    lateinit var radioGroup: RadioGroup
    lateinit var linearLayout:LinearLayout
    lateinit var sariRadioButton:RadioButton
    lateinit var kirmiziRadioButton: RadioButton
    lateinit var yesilRadioButton: RadioButton
    lateinit var cyanRadioButton: RadioButton
    lateinit var listem:ListView
    lateinit var ratingDegisken:RatingBar
    lateinit var ratinSonuc:TextView

    val yemekler=listOf("Kuru Fasulye","Lahmacun","Pilav","Kebap","Patates Kızartması","İmambayıldı")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        secim=findViewById(R.id.secim)
        radioGroup=findViewById(R.id.renkler)
        linearLayout=findViewById(R.id.main)
        sariRadioButton=findViewById(R.id.sari)
        kirmiziRadioButton=findViewById(R.id.kirmizi)
        yesilRadioButton=findViewById(R.id.yesil)
        cyanRadioButton=findViewById(R.id.cyan)
        listem=findViewById(R.id.listem)
        ratingDegisken=findViewById(R.id.ratingBar)
        ratinSonuc=findViewById(R.id.ratingSonuc)

        val adapter=ArrayAdapter<String>(this,android.R.layout.simple_list_item_1,yemekler)
        listem.adapter=adapter
        listem.onItemClickListener=AdapterView.OnItemClickListener{parent,view,position,id->
            val secilenYemek=parent.getItemAtPosition(position) as String
            Toast.makeText(this,"Seçilen yemek= $secilenYemek",Toast.LENGTH_LONG).show()
        }

        secim.setOnClickListener{
            /*
            var secilenRadioButton=radioGroup.checkedRadioButtonId
            when(secilenRadioButton){
                R.id.kirmizi->linearLayout.setBackgroundColor(Color.RED)
                R.id.yesil->linearLayout.setBackgroundColor(Color.GREEN)
                R.id.sari->linearLayout.setBackgroundColor(Color.YELLOW)
                R.id.cyan->linearLayout.setBackgroundColor(Color.CYAN)
            }
             */

            val color=Color.rgb(Random.nextInt(256),Random.nextInt(256),Random.nextInt(256))
            linearLayout.setBackgroundColor(color)
        }

        kirmiziRadioButton.setOnClickListener{
            linearLayout.setBackgroundColor(Color.RED)
        }
        yesilRadioButton.setOnClickListener{
            linearLayout.setBackgroundColor(Color.GREEN)
        }
        sariRadioButton.setOnClickListener{
            linearLayout.setBackgroundColor(Color.YELLOW)
        }
        cyanRadioButton.setOnClickListener{
            linearLayout.setBackgroundColor(Color.CYAN)
        }

        ratingDegisken.setOnRatingBarChangeListener { ratingBar, rating, fromUser ->
            ratinSonuc.text="Puan = $rating"
        }
    }
}